packages/site/cms-site/src/blockRegistry/blockRegistry.ts

```diff
@@ -74,11 +75,13 @@ export async function recursivelyLoadBlockData({
     blockData,
     client,
     blocksMeta,
+    cache,
 }: {
     blockType: string;
     blockData: unknown;
     client: GraphQLClient;
```

Comment:
@nsams we still have a dependency on the client in the library here, right?

Have you thought about not passing the client around, and instead pass the visibility options (`previewDamUrls` etc.) instead? This would allow defining the GraphQL client in the loader (a few createGraphQLClient calls won't hurt) and remove the dependency.

Reply:
Inspired by https://github.com/vivid-planet/comet/pull/1596#discussion_r1502536765 I removed the dependency and replaced it with a generic type: c7269974f3a9dc986a21a7f057c461dbb52d1cad

Reply:
> Have you thought about not passing the client around, and instead pass the visibility options (previewDamUrls etc.) instead?

What do you think about this? We'd presumably need this for rest requests to the API, e.g. a dam file url

Reply:
> not passing the client around, and instead pass the visibility options

hm, I like the pattern of having a single client instance that you configure once. It is also known by apollo client.

> We'd presumably need this for rest requests to the API, e.g. a dam file url

I don't understand that

Reply:
> I don't understand that

You proposed the possibility to not use GraphQL, but fetch from a third party directly in the loader. I thought maybe we'd need to fetch something from our API using REST, where we'd also need to consider the visibility options (i.e. setting the x-include-invisible-content header for the fetch call). But maybe that's a contrived example.